test/integration/asyncprofiler_test.go

```diff
@@ -0,0 +1,138 @@
+	javaPID := cmd.Process.Pid
+	fmt.Fprintf(os.Stdout, "Java PID: %d\n", javaPID)
+
+	// Initialize AsyncProfiler with required parameters
+	profiler := asyncprofiler.NewAsyncProfiler(
+		filepath.Join(cwd, "testdata", "jattach"),
+		filepath.Join(cwd, "testdata", "libasyncProfiler.so"),
+		javaPID,
+	)
+
+	// Run fdtransfer
+	err = asyncprofiler.RunFdtransfer(filepath.Join(cwd, "testdata", "fdtransfer"), javaPID)
+	require.NoError(t, err, "Failed to run fdtransfer")
+
+	time.Sleep(10 * time.Second) // Wait for fdtransfer to finish loading
+
+	jfrOutputFile := filepath.Join(cwd, "prof.jfr")
+	// Set action to start
+	err = profiler.SetAction("start", asyncprofiler.ProfilerOptions{
+		"event": "cpu",
+		"file":  jfrOutputFile,
+	})
+	require.NoError(t, err, "Failed to set action")
+
+	// Construct the start command
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+	profilerCmd, err := profiler.BuildCommand(ctx)
+	require.NoError(t, err, "Failed to build command")
+
+	// Execute the start command
+	var outbuf, errbuf bytes.Buffer
+	profilerCmd.Stdout = &outbuf
+	profilerCmd.Stderr = &errbuf
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", profilerCmd)
+	if err = profilerCmd.Run(); err != nil {
+		err = fmt.Errorf("Failed to execute profiler: %w: %s: %s", err, outbuf.String(), errbuf.String())
+	}
+	require.NoError(t, err, "Failed to run start command")
+
+	// Sleep for the specified duration
+	time.Sleep(120 * time.Second)
+
+	// Set action to stop
+	err = profiler.SetAction("stop")
+	require.NoError(t, err, "Failed to set action")
+
+	// Construct the stop command
+	ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	profilerCmd, err = profiler.BuildCommand(ctx)
+	require.NoError(t, err, "Failed to build command")
+
+	// Execute the stop command
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", profilerCmd)
+	err = profilerCmd.Run()
+	require.NoError(t, err, "Failed to run stop command")
+
+	// Check the exit code
+	require.Equal(t, 0, profilerCmd.ProcessState.ExitCode(), "Non-zero exit code")
+
+	// Check if the output file "prof.jfr" exists
+	_, err = os.Stat(jfrOutputFile)
+	require.NoError(t, err, "Failed to find output file")
+
+	jfrFile, err := os.Open(jfrOutputFile)
+	require.NoError(t, err, "Failed to open JFR file")
+	defer jfrFile.Close()
+
+	pprofProfile, err := convert.JfrToPprof(jfrFile)
+	require.NoError(t, err, "Failed to convert JFR to pprof")
+	require.NotNil(t, pprofProfile, "Pprof profile is nil")
+
+	// Save the converted pprof profile to a file
+	pprofFile, err := os.Create("prof.pprof")
+	require.NoError(t, err, "Failed to create pprof file")
+	defer pprofFile.Close()
+
+	err = pprofProfile.Write(pprofFile)
+	require.NoError(t, err, "Failed to write pprof profile")
+
+	// Clean up the output files
+	outputFiles := []string{jfrOutputFile, "prof.pprof"}
+
+	for _, outputFile := range outputFiles {
+		err = os.Remove(outputFile)
```

Comment:
the typical thing to do here is a `defer os.Remove(file)` right after creating the file just like it's already done with the `.Close()`.